source/NVDAObjects/IAccessible/__init__.py

```diff
@@ -249,9 +250,14 @@ def _getCharacterOffsets(self,offset):
 		except COMError:
 			pass
 		try:
-			return self.obj.IAccessibleTextObject.TextAtOffset(offset,IAccessibleHandler.IA2_TEXT_BOUNDARY_CHAR)[0:2]
+			start,end,text = self.obj.IAccessibleTextObject.TextAtOffset(offset,IAccessibleHandler.IA2_TEXT_BOUNDARY_CHAR)

```

Comment:
Although I understand the advantage of the changes in this file, I don't quite see how it fits in this pr, especially around display of braille... If it is unrelated, please remove it.

Reply:
May be having a look at braille.TextInfoRegion.getTextInfoForBraillePos makes this clearer.

In short, when using cursor routing keys in stable versions of NVDA within IA2 text controls with emoji, cursor routing works perfectly. This is because a surrogate pair takes two characters/offsets in IA2TextInfo as well as in brailleToRawPos. If moving to UCS4 however, surrogate pairs only take one index in brailleToRawPos. Therefore, moving to UCS4 for liblouis introduces off by one errors when using cursor routing in IA2text with emoji. The same would happen if using this particular change to master without moving to UCS4. Therefore, I consider these two cases fairly related. If you still think otherwise, I'm happy to split the pr though, but they still need to be merged one after the other in order not to introduce a regression.